### ISSUE TITLE
Reading/writing from/to multiple locations in one command

### ISSUE THREAD

**Issue by doesthisusername** [xemio/ps4debug#23]
Hi,
I've been getting back into this sort of stuff a little bit recently, and am making an "information display" of sorts for a game, where I'm hoping to display various information read from memory, ideally at 30 Hz or higher. I don't think I can do that currently if I have a lot of synchronous read commands, so I'm wondering if you agree that a new command with [iov-like](https://man7.org/linux/man-pages/man2/process_vm_readv.2.html) capabilities would be useful (basically doing what the title says).

If so, I'm not sure what the best implementation would be. In my head right now, I'm imagining something like how process lists are returned currently, where the requester would send something like a pid and amount of records it's going to send, followed by the records themselves. I'm also thinking about the scope of this, as a current use-case for reading memory is pointer path traversal, which you can't do in a single request like this, unless you specifically account for it.

Writing to memory might also be useful, but that then adds length variations.

Lastly, I have a few more questions/things (unable to get ELF loading to work, a few C++ libdebug bugs, am currently porting libdebug to [Zig](https://ziglang.org)), but am unsure about whether this is the proper place to post them.

btw I'm willing to implement this in a PR if you'd like, just wanted to run it by you first.
Thanks!

**Comment by jogolden** [comment]
Yes I would love a PR, if you are working on 5.05! Sorry for the late reply. I dont really maintain or work on PS4 public projects anymore.

I was not the author of the C++ libdebug @ChendoChap and honestly Ive never used it. I know it used experimental C++ features (maybe not experimental anymore). I do wonder why ELF loading is not working for you.

**Comment by doesthisusername** [comment]
Cool, yeah I'm on 5.05. Unfortunately I'm not working on this project at the moment, but I'll keep this in mind if I go back to it.
